eth/handler.go

```diff
@@ -372,6 +372,8 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		if err := msg.Decode(&query); err != nil {
 			return errResp(ErrDecode, "%v: %v", msg, err)
 		}
+		hashMode := query.Origin.Hash != (common.Hash{})
```

Comment:
That's also possible and actually was my original choice. The reason I moved it above is because the var declaration is already part of the query resolution (i.e. header chain iteration) while this part is kind of an initialization code to define *how* the iteration works. Seemed a bit nicer logically, but I can change it if you consider it better inside the var.